modules/secure-preview/lib/secure-preview.js

```diff
@@ -0,0 +1,240 @@
+			return result.url;
+		}
+	};
+
+	const handleUrlCopied = url => {
+		dispatch( noticesStore ).createNotice( 'info', __( 'Link copied to clipboard.' ), {
+			isDismissible: true,
+			type: 'snackbar',
+		} );
+
+		onCloseModal();
+		onUrl( url );
+	};
+
+	const selectOptions = expirationOptions.map( ( { label, value } ) => {
+		return { label, value };
+	} );
+
+	return (
+		<Modal title="Generate secure preview" size="medium" onRequestClose={ onCloseModal }>
+			<SelectControl
+				label={ __( 'Link expiration', 'vip-workflow' ) }
+				value={ expiration }
+				onChange={ setExpiration }
+				options={ selectOptions }
+			/>
+
+			<CheckboxControl
+				className="vip-workflow-one-time-use-checkbox"
+				label={ __( 'One-time use', 'vip-workflow' ) }
+				help={ __( 'The link will expire after one visit.', 'vip-workflow' ) }
+				checked={ isOneTimeUse }
+				onChange={ () => setIsOneTimeUse( isOneTimeUse => ! isOneTimeUse ) }
+			/>
+
+			<Flex justify="flex-end">
+				{ isLoading && <Spinner /> }
+
+				<CopyFromAsyncButton
+					variant="primary"
+					asyncFunction={ getSecureUrl }
+					onCopied={ handleUrlCopied }
+				>
+					{ __( 'Copy Link', 'vip-workflow' ) }
+				</CopyFromAsyncButton>
+			</Flex>
+		</Modal>
+	);
+};
+
+const SecurePreviewDropdown = ( { url } ) => {
+	const anchorRef = useRef( null );
+
+	const clipboardRef = useCopyToClipboard( url, () => {
+		dispatch( noticesStore ).createNotice( 'info', __( 'Link copied to clipboard.' ), {
+			isDismissible: true,
+			type: 'snackbar',
+		} );
+	} );
+
+	const popoverProps = {
+		anchorRef,
+		placement: 'left-start',
+		offset: 36,
+		shift: true,
+	};
+
+	// Grab everything from the last '/' in the URL
+	const urlTail = '/' + url.split( '/' ).pop();
+
+	return (
+		<div className="vip-workflow-secure-preview-dropdown" ref={ anchorRef }>
+			<Dropdown
+				popoverProps={ popoverProps }
+				focusOnMount
+				renderToggle={ ( { onToggle } ) => (
+					<Button size="compact" variant="tertiary" onClick={ onToggle }>
+						<Truncate limit={ 15 } ellipsizeMode="tail">
+							{ urlTail }
+						</Truncate>
+					</Button>
+				) }
+				renderContent={ ( { onClose } ) => (
+					<div className="vip-workflow-secure-link-dropdown-content">
+						<InspectorPopoverHeader title={ __( 'Secure Preview Link' ) } onClose={ onClose } />
+						<ExternalLink className="editor-post-url__link" href={ url } target="_blank">
+							{ url }
+						</ExternalLink>
+					</div>
+				) }
+			/>
+
+			<Button
+				icon={ copySmall }
+				label={ sprintf(
+					// Translators: %s is a placeholder for the link URL, e.g. "Copy link: https://my-site.com/?p=123".
+					__( 'Copy link: %s' ),
+					url
+				) }
+				ref={ clipboardRef }
+				size="compact"
+			/>
+		</div>
+	);
+};
+
+const getPostProperties = select => {
+	const { getEditedPostAttribute, getCurrentPostType, getCurrentPost } = select( editorStore );
+
+	const post = getCurrentPost();
+	const isDraft = post?.status === 'auto-draft';
```

Comment:
I'd a comment here explaining why this is done this way, and change the name of this var to be `isAutoDraft` instead.

Reply:
Agree that `isDraft` is a confusing name, especially given that we have a "Draft" status. I changed it to `isUnsavedPost` in https://github.com/Automattic/vip-workflow-plugin/pull/19/commits/ac739e071f6a077d3c63728c42cf1b639149d23c which makes the usage clearer in the plugin, and also added a comment.